src/rust/bitbox02-rust/src/hww/api/ethereum/sign.rs

```diff
@@ -231,22 +357,16 @@ pub async fn process(request: &pb::EthSignRequest) -> Result<Response, Error> {
         }
     }
 
-    let hash = super::sighash::compute(&super::sighash::Params {
-        nonce: &request.nonce,
-        gas_price: &request.gas_price,
-        gas_limit: &request.gas_limit,
-        recipient: &recipient,
-        value: &request.value,
-        data: &request.data,
-        chain_id: params.chain_id,
-    })
-    .or(Err(Error::InvalidInput))?;
+    let hash: [u8; 32] = match request {
+        Transaction::Legacy(legacy) => hash_legacy(params.chain_id, recipient, legacy)?,
+        Transaction::Eip1559(eip1559) => hash_eip1559(params.chain_id, recipient, eip1559)?,
+    };
 
-    let host_nonce = match request.host_nonce_commitment {
+    let host_nonce = match request.host_nonce_commitment() {
```

Comment:
for Eip1559, please force the use of anti-klepto. No need to make it optional because there are backwards compatibility concerns as with legacy, where anti-klepto was introduced after wallets had already been using the non-anti-klepto signing.

Reply:
nevermind - the client libraries can/should enforce antiklepto for eip1559, the firmware does not have to. you can also enforce it in the firmware if it is eay, but that is not so important.